crates/cxx-qt-lib-extras/src/gui/qapplication.rs

```diff
@@ -0,0 +1,200 @@
+        #[rust_name = "qapplication_application_name"]
+        fn qapplicationApplicationName(app: &QApplication) -> QString;
+        #[doc(hidden)]
+        #[rust_name = "qapplication_remove_library_path"]
+        fn qapplicationRemoveLibraryPath(app: &QApplication, path: &QString);
+        #[doc(hidden)]
+        #[rust_name = "qapplication_application_version"]
+        fn qapplicationApplicationVersion(app: &QApplication) -> QString;
+        #[doc(hidden)]
+        #[rust_name = "qapplication_exec"]
+        fn qapplicationExec(app: Pin<&mut QApplication>) -> i32;
+        #[doc(hidden)]
+        #[rust_name = "qapplication_library_paths"]
+        fn qapplicationLibraryPaths(app: &QApplication) -> QStringList;
+        #[doc(hidden)]
+        #[rust_name = "qapplication_organization_domain"]
+        fn qapplicationOrganizationDomain(app: &QApplication) -> QString;
+        #[doc(hidden)]
+        #[rust_name = "qapplication_organization_name"]
+        fn qapplicationOrganizationName(app: &QApplication) -> QString;
+        #[doc(hidden)]
+        #[rust_name = "qapplication_set_application_name"]
+        fn qapplicationSetApplicationName(app: Pin<&mut QApplication>, name: &QString);
+        #[doc(hidden)]
+        #[rust_name = "qapplication_set_application_version"]
+        fn qapplicationSetApplicationVersion(app: Pin<&mut QApplication>, version: &QString);
+        #[doc(hidden)]
+        #[rust_name = "qapplication_set_font"]
+        fn qapplicationSetFont(app: Pin<&mut QApplication>, font: &QFont);
+        #[doc(hidden)]
+        #[rust_name = "qapplication_font"]
+        fn qapplicationFont(app: &QApplication) -> QFont;
+        #[doc(hidden)]
+        #[rust_name = "qapplication_set_library_paths"]
+        fn qapplicationSetLibraryPaths(app: Pin<&mut QApplication>, paths: &QStringList);
+        #[doc(hidden)]
+        #[rust_name = "qapplication_set_organization_domain"]
+        fn qapplicationSetOrganizationDomain(app: Pin<&mut QApplication>, domain: &QString);
+        #[doc(hidden)]
+        #[rust_name = "qapplication_set_organization_name"]
+        fn qapplicationSetOrganizationName(app: Pin<&mut QApplication>, name: &QString);
+    }
+
+    // QApplication is not a trivial to CXX and is not relocatable in Qt
+    // as the following fails in C++. So we cannot mark it as a trivial type
+    // and need to use references or pointers.
+    // static_assert(QTypeInfo<QApplication>::isRelocatable);
+    impl UniquePtr<QApplication> {}
+}
+
+pub use ffi::QApplication;
+
+impl QApplication {
+    /// Prepends path to the beginning of the library path list,
+    /// ensuring that it is searched for libraries first.
+    /// If path is empty or already in the path list, the path list is not changed.
+    pub fn add_library_path(self: Pin<&mut Self>, path: &QString) {
+        ffi::qapplication_add_library_path(self, path);
+    }
+
+    /// The name of this application
+    pub fn application_name(&self) -> QString {
```

Comment:
don't some of these come from `QCoreApplication` or `QGuiApplication`.

It would be good if we figured out a plan with @LeonMatthesKDAB around how to handle inheritance, we were considering if we just implement From/Into on these Qt classes. Eg QStringList -> QList<QString> too so that we don't have to implement the methods again for all types.

Reply:
Up and downcast C++ helpers was in #823 not sure we have an issue to track From/Into or a trait for Deref or something yet.

Reply:
indeed allow to handle inheritance will be good. 
I don't know how we can do at the moment.
but indeed better !

Reply:
Right, this needs to be discussed, we have a few ideas :-) Basically have some kind of trait or use From/Into on the Rust side to call the cast on the C++ side. Then it'll return the other type. (and use TryFrom/TryInto where it can be null etc). But lets figure out a proper plan with @LeonMatthesKDAB :-)

Reply:
Lets investigate this in a later change as part of #562